### ISSUE TITLE
class which extends directly from Object, always returns as an Object instance

### ISSUE THREAD

**Issue by dqweek** [Microsoft/TypeScript#12905]
**TypeScript Version:**  2.1.1 / nightly (2.2.0-dev.201xxxxx)

**Code**

```ts

class Main extends Object {
}

const main = new Main();

const isMain: boolean = main instanceof Main; <-- should be true

```

**Expected behavior:**

isMain variable should be true
or 
Compiler should emit an error that extending directly from Object is not allowed

**Actual behavior:**

isMain = false;
actual type is Object

**Comment by HerringtonDarkholme** [comment]
https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work

**Comment by dqweek** [comment]
Thnx! 
Missed that part :)

closing
